### DISCUSSION TITLE
Run Test Scripts Without Running New Request

### DISCUSSION THREAD

**Discussion by JasonNham** [usebruno/bruno#1449] [Ideas]
One of the things I noticed when testing Bruno is there doesn't appear to be a way to update and run tests without running a new api request. Not that this is a particularly huge issue for independent testing but I was wondering if this is a planned feature for the future?

**Comment by helloanoop** [comment]
Hi @JasonNham Can you describe your use-case a bit more ?

**Reply by JasonNham** [reply]
Hi, thank you for your response. I’m currently exploring the possibility of caching API responses for testing purposes. Specifically, I’m looking for a feature that allows me to save or cache the response from an API request until the next one is executed.

The reason behind this is that I’m developing test scripts for my API requests. When I initially write these scripts, I need to send another API request to test them. It would be more efficient if I could run these scripts against a previously received response. This way, I can avoid making multiple requests solely for testing my scripts.

Is there an existing feature that supports this, or is there one in development? Your guidance would be greatly appreciated.

**Comment by Nsbx** [comment]
Hey,
Sorry for digging this up, but I had the same need just now and came across this thread.
In my case, I write my tests and requests at the same time as I write my API.
And in the case of methods that modify, create, or destroy data (POST, DELETE, PUT), I would like to rerun the tests without having to clean my DB with each call, just for the time it takes to write the tests.
